src/candid.ts

```diff
@@ -0,0 +1,163 @@
+): Promise<CandidPremierResult> => {
+  logger.debug(`Looking up EIN ${ein} in Candid Premier API`);
+  const response = await client.get<CandidPremierResult>(
+    `https://api.candid.org/premier/v3/${ein}`,
+    {
+      headers: {
+        'Subscription-Key': apiKey,
+      },
+    },
+  );
+  logger.debug(`Fetched Candid data for ${ein}: ${response.data.message}`);
+  return response.data;
+};
+
+const getEinBaseFieldId = async (
+  baseUrl: string,
+  token: AccessTokenSet,
+) => {
+  const baseFields = await getBaseFields(baseUrl, token);
+  const einBaseField = baseFields.find(({ shortCode }) => (
+    shortCode === 'organization_tax_id'
+  ));
+  if (einBaseField === undefined) {
+    throw new Error('Could not find base field with short code `organization_tax_id`');
+  }
+  return einBaseField.id;
+};
+
+const isValidEin = (s: string): boolean => (
+  /^\d{2}-?\d{7}$/.test(s)
+);
+
+const getEinsFromPdc = async (baseUrl: string, token: AccessTokenSet) => {
+  const einBaseFieldId = await getEinBaseFieldId(baseUrl, token);
+  const proposals = await getProposals(baseUrl, token);
+
+  const eins = new Set(proposals.entries
+    .flatMap((proposal) => proposal.versions[0]?.fieldValues)
+    .filter(<T>(x: T | undefined): x is T => typeof x !== 'undefined')
+    .filter(({ applicationFormField }) => applicationFormField.baseFieldId === einBaseFieldId)
+    .map(({ value }) => value)
+    .filter(isValidEin));
+  return [...eins];
+};
+
+const candid: CommandModule = {
+  command: 'candid',
+  describe: 'Interact with the Candid Premier API',
+  builder: (y) => (y
+    .option('candid-api-key', {
+      describe: 'Candid Premier API key; get from account management at https://dashboard.candid.org/',
+      demandOption: true,
+      global: true,
+      type: 'string',
+    })
+    .command(
+      'lookup <ein>',
+      'Fetch and display information about an organization by its EIN',
+      {
+        'output-file': {
+          alias: 'write',
+          describe: 'Write organization information to the specified JSON file',
+          normalize: true,
+          type: 'string',
+        },
+      },
+      async (args) => {
+        const result = await getCandidProfile(
+          args.candidApiKey as string,
+          args.ein as string,
+        );
+
+        if (typeof args.outputFile === 'string') {
+          await writeFile(
+            args.outputFile,
+            JSON.stringify(result, null, 2),
+          );
+          logger.info(`Wrote Candid data for ${args.ein as string} to ${args.outputFile}`);
+        } else {
+          logger.info('Candid result', { result });
+        }
+      },
+    )
+    .command(
+      'update <ein>',
+      'Fetch information about an organization by its EIN, and upload to the PDC',
+      oidcOptions,
+      async (args) => {
+        const token = await getToken(
+          args.oidcBaseUrl as string,
+          args.oidcClientId as string,
+          args.oidcClientSecret as string,
```

Comment:
I'm not a fan of this, nor of the sequence of type guards alternative, but either the types for yargs or TypeScript itself doesn't allow for statically typing these slightly-dynamic options. If you have any suggestions for best to tell TypeScript that "hey, this can't be `undefined`, actually", I'd love to hear them!

Reply:
I'm not sure I understand the problem space, but does this help? https://github.com/yargs/yargs/blob/main/docs/typescript.md#typescript-usage-examples